google-home-community.groovy

```diff
@@ -562,6 +578,126 @@ private deviceTraitPreferences_Dock(deviceTrait) {
     }
 }
 
+@SuppressWarnings(['UnusedPrivateMethod', 'MethodSize'])
+private deviceTraitPreferences_EnergyStorage(deviceTrait) {
+    googleEnergyStorageDistanceUnitForUX = [
+        "KILOMETERS":      "Kilometers",
+        "MILES":           "Miles",
+    ]
+    googleCapacityUnits = [
+        "SECONDS":         "Seconds",
+        "MILES":           "Miles",
+        "KILOMETERS":      "Kilometers",
+        "PERCENTAGE":      "Percentage",
+        "KILOWATT_HOURS":  "Kilowatt Hours",
+    ]
+    section("Energy Storage Settings") {
+        input(
+            name: "${deviceTrait.name}.isRechargeable",
+            title: "Rechargeable",
+            type: "bool",
+            defaultValue: false,
+            required: true,
+            submitOnChange: true
+        )
+        input(
+            name: "${deviceTrait.name}.capacityRemainingRawValue",
+            title: "Capacity Remaining Value",
+            type: "text",
+            defaultValue: "battery",
+            required: true
+        )
+        input(
+            name: "${deviceTrait.name}.capacityRemainingUnit",
+            title: "Capacity Remaining Unit",
+            type: "enum",
+            options: googleCapacityUnits,
+            defaultValue: "PERCENTAGE",
+            multiple: false,
+            required: true,
+            submitOnChange: true
+        )
+        input(
+            name: "${deviceTrait.name}.advancedEnergySettings",
+            title: "Enable Advanced Settings",
+            type: "bool",
+            defaultValue: false,
+            submitOnChange: true
+        )
+    }
+    if (deviceTrait.advancedEnergySettings) {
+        section("Advanced Settings") {
```

Comment:
Rather than a setting you should use the `hidable` and `hidden` arguments to the `section` function:
```
section(hidable: true, hidden: true, "Advanced Settings")
```

Reply:
Sorry, missed your note, and you replied before I could implement.  :)

I've changed it to the hideable section.

Reply:
Is there a class definition for section that would format the first section the same, but not be hideable?

Reply:
You can use `hideable` without `hidden` to make them look the same but not be collapsed by default.  I think it's better to leave the basic settings non-hideable though.

Reply:
Gotcha, I'll leave it as is.